modules/tinymce/src/core/main/ts/NodeChange.ts

```diff
@@ -104,20 +105,21 @@ class NodeChange {
       }
 
       // Get parents and add them to object
-      parents = [];
+      const parents: Node[] = [];
       this.editor.dom.getParent(node, (node) => {
         if (node === root) {
           return true;
+        } else {
+          parents.push(node);
+          return false;
         }
-
-        parents.push(node);
       });
 
-      args = args || {};
-      args.element = node;
-      args.parents = parents;
-
-      this.editor.dispatch('NodeChange', args);
+      this.editor.dispatch('NodeChange', {
+        ...args,
+        element: node,
+        parents
+      });
```

Comment:
More prominent code change here, as the old code here mutated the `args` passed in, whereas this clones it now.